src/components/pages/TransactionHistoryPage.js

```diff
@@ -1,34 +1,106 @@
-import { React, useState, useEffect } from "react";
+import { React, useState, useEffect, useContext, useCallback } from "react";
 import Navbar from "../reusable/NavBar";
 import Sidebar from "../reusable/SideBar";
+import AlgoquantApiContext from "../../api/ApiContext";
+import Table from "../reusable/Table";
+import formatter from "../utils/CurrencyFormatter";
+import { TableSpinner } from "../reusable/LoadSpinner";
 
+const FETCH_AMOUNT = 10;
 const TransactionHistoryPage = () => {
+  const [history, setHistory] = useState([]);
+  const [lastPage, setLastPage] = useState(false);
   const [transactions, setTransactions] = useState([]);
   const [page, setPage] = useState(1);
+  // State variables used to access algoquant SDK API and display/ keep state of user data from database
+  const algoquantApi = useContext(AlgoquantApiContext);
+  const [lastKey, setLastKey] = useState(null);
+  const [pagesSeen, setPagesSeen] = useState(0);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchTrades = useCallback(
+    (FETCH_AMOUNT) => {
+      const historyBuffer = [];
+      if (!lastPage && pagesSeen < page) {
+        if (algoquantApi.token) {
+          algoquantApi
+            .getTrades(FETCH_AMOUNT, lastKey)
+            .then((resp) => {
+              setLastKey(null);
```

Comment:
Do you need to set this to null here ?

Reply:
this we do because because i want it to reset after the last query, or the last query's "last evaluated key" presist in the state. i could do an else statement if u want.